src/sardana/macroserver/macros/standard.py

```diff
@@ -470,20 +472,32 @@ class mv(Macro):
     ]
 
     def run(self, motor_pos_list):
-        motors, positions = [], []
+        enable_hooks = getattr(sardanacustomsettings,
+                               'PRE_POST_MOVE_HOOK_IN_MV',
+                               True)
+
+        if enable_hooks:
+            for preMoveHook in self.getHooks('pre-move'):
+                preMoveHook()
+
+        self.motors, positions = [], []
         for m, p in motor_pos_list:
-            motors.append(m)
+            self.motors.append(m)
```

Comment:
We need to fill the `self.motors` before executing `pre-move` hooks. They may need to access to it.

In this case I would make two iterations. In the first one fill the `self.motors` and in the second one print the debug message. IMHO the debug message should stay as close as possible to the move itself.